### ISSUE TITLE
Problems with Issues search functionality

### ISSUE THREAD

**Issue by KazzmanK** [go-gitea/gitea#27822]
### Description

Following #27722 I decided to create separate issue for adding dependencies.

I have an Organization with several repositories.
Repository 1 have Issue R1_I1 and R1_I2
Repository 2 have Issue R2_I1 and R2_I2
R1_I2 depends on R1_I1
R2_I2 depends on R2_I1
and 
R2_I1 depends on R1_I1
R2_I2 depends on R1_I2

While adding in-repository dependencies it is okay, but when I try to add dependency for cross-repository issues, required issue is just not shown in dropdown.

When I want to add R1_I1 as dependency for R2_I1 filter says nothing found, but if I wand to add R1_I2 as dependency, it magically displays R2_I1 .
Using this trick I can add one dependency, but not the other.

Overall , in dependency dropdown some kind on random results are shown.

And I cannot reproduce this kind of problem on try.gitea.io
but on https://try.gitea.io/n.kobzarev/front_repo/issues/2 I'm unable to remove "back_issue_feature_1", and it is not readable for unknown reason.

### Gitea Version

1.22.0+dev-249-ge75b89ea6

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![image](https://github.com/go-gitea/gitea/assets/13328513/63f12abb-629a-450d-aceb-c1bcb4b94e3f)

### How are you running Gitea?

windows service

### Database

SQLite

**Comment by KazzmanK** [comment]
Finally, I`ve managed to find the root cause, I hope.

Lets find some issues in https://gitea.local/issues?

- Daylight - issue founded (number `#160`)
- daylight - same founded, so latin characters are case insensitive
- dayligh - NOT FOUNDED by partial word
- at least non latin full words are case sensitive in case of using Issue indexer (Brave)
- `#160` - NOT FOUNDED , must have functionality to refer issues by ID
- `#32` - FOUNDED some non relevant issues, where substring 32 is a part of screenshot file name (GUID)
Last part is disappointing, because searching for part of real word fails, but by part of guid is not.

So to successfully add dependencies in Issues I need to know exact issue name (or part of comment inside issue), not just ID, even then there is a chance that required issue will be missing in dropdown

**Comment by Jaffex** [comment]
I had the same issue today - at first I didn't think it was even possible to create a dependency to an issue in a different repository, but then I noticed the search already suggests some other issues from other repos.

The thing that threw me off track was that the dependency search was not able to identify the issue I wanted to link even when I copied and pasted the "Reference"-value from the target issue.

Only when I started to enter text from the target issue's comment did the right issue show up.
